### DISCUSSION TITLE
Best practices for Hooks?

### DISCUSSION THREAD

**Discussion by jwyse** [garethgeorge/backrest#600] [Q&A] [answered]
Since both Plans and Repositories support Hooks, are there any recommended best practices, or pros/cons, for where to define hooks?

I have 3 plans and 3 corresponding repositories: PlanA->RepoA, PlanB->RepoB, PlanC->RepoC.  I have Hooks for Shoutrrr notifications on `CONDITION_SNAPSHOT_END` and `CONDITION_ANY_ERROR` events, currently defined within the plans.  Should I reconfigure the notifications at the Repo level to ensure that I get notifications for Prune/Check errors, or would the Plan hooks catch those?

**Comment by garethgeorge** [comment, marked as answer]
This is a good question

* Plan level hooks are triggered by backup and forget operations
* Repo level hooks are triggered by any plan level operations running on the repo + prune and check operations

If you're wanting notifications for system health events you should have those hooks configured at the repo level.

The easiest way to move them might be to just stop backrest and manually edit the config, you can copy the json "hooks" section from your plans to your repos if you're comfortable editing the file -- the field name and properties are the same.
